### PR TITLE
chore: code split for consolidated API

### PR DESCRIPTION
**Pull request by AnaghHegde**
## Description
This PR does the code split of consolidated API to support adding the environment details consolidated API in EE, as mentioned here in the issue. - https://github.com/appsmithorg/appsmith/issues/33132

Fixes https://github.com/appsmithorg/appsmith/issues/33132

## Automation

/ok-to-test tags="@tag.Perf"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/9595115979>
> Commit: 7e9cbe0a28807dec5ef3df168698508dbe9e0d0c
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=9595115979&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Perf`

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
  - Introduced a consolidated API response structure to streamline data from various endpoints, including user profiles, feature flags, and more.
  - Added comprehensive support for loading consolidated page information, enhancing the efficiency of page load operations.
  
- **Refactor**
  - Refactored internal service interfaces and implementations to improve maintainability and extendability.
  - Centralized the `getConsolidatedInfoForPageLoad` method for better code organization and usability.

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ConsolidatedAPIService.java [hunk 1]
```diff
@@ -1,10 +1,5 @@
...
-import com.appsmith.server.domains.ApplicationMode;
-import com.appsmith.server.dtos.ConsolidatedAPIResponseDTO;
-import reactor.core.publisher.Mono;
+import com.appsmith.server.services.ce.ConsolidatedAPIServiceCE;
 
-public interface ConsolidatedAPIService {
-    Mono<ConsolidatedAPIResponseDTO> getConsolidatedInfoForPageLoad(
-            String defaultPageId, String applicationId, String branchName, ApplicationMode mode);
-}
+public interface ConsolidatedAPIService extends ConsolidatedAPIServiceCE {}
```
Comment (abhvsn) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ConsolidatedAPIService.java:5]:
```suggestion
public interface ConsolidatedAPIService extends ConsolidatedAPIServiceCECompatible {}
```

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ConsolidatedAPIServiceImpl.java [hunk 2]
```diff
@@ -1,430 +1,54 @@
...
-                    .name(getQualifiedSpanName(MOCK_DATASOURCES_SPAN, mode))
-                    .tap(Micrometer.observation(observationRegistry)));
-        }
-
-        return Mono.when(fetches).thenReturn(consolidatedAPIResponseDTO);
-    }
-
-    private boolean isPossibleToCreateQueryWithoutDatasource(Plugin plugin) {
-        return PLUGINS_THAT_ALLOW_QUERY_CREATION_WITHOUT_DATASOURCE.contains(plugin.getPackageName());
+public class ConsolidatedAPIServiceImpl extends ConsolidatedAPIServiceCEImpl implements ConsolidatedAPIService {
```
Comment (abhvsn) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ConsolidatedAPIServiceImpl.java:18]:
```suggestion
public class ConsolidatedAPIServiceImpl extends ConsolidatedAPIServiceCECompatibleImpl implements ConsolidatedAPIService {
```

**Review by abhvsn — COMMENTED** [review]
Minor changes but looks good

**Comment by abhvsn** [comment]
@AnaghHegde build is failing can you take a look at this once.

**Comment by AnaghHegde** [comment]
@abhvsn The commits i made is not visible for some reason in the PR.

**Comment by AnaghHegde** [comment]
@abhvsn Created new PR here- https://github.com/appsmithorg/appsmith/pull/34356. The copmmits are not available in this for reason i don't know.

**Review by abhvsn — APPROVED** [review]

**Pull request merged**